profiler/export_profiles/main.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START profiler_export_profiles]
+
+// Sample export_profiles shows how ListProfiles API can be used to download
+// existing pprof profiles for a given project from GCP.
+package main
+
+import (
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	cloudprofiler "cloud.google.com/go/cloudprofiler/apiv2"
+	pb "cloud.google.com/go/cloudprofiler/apiv2/cloudprofilerpb"
+	"google.golang.org/api/iterator"
+)
+
+var project = flag.String("project", "", "GCP project ID from which profiles should be fetched")
+var pageSize = flag.Int("page_size", 100, "Number of profiles fetched per page. Maximum 1000.")
+var pageToken = flag.String("page_token", "", "PageToken from a previous ListProfiles call. If empty, the listing will start from the begnning. Invalid page tokens result in error.")
+var maxProfiles = flag.Int("max_profiles", 1000, "Maximum number of profiles to fetch across all pages. If this is <= 0, will fetch all available profiles")
+
+// This function reads profiles for a given project and stores them into locally created files.
+// The profile metadata gets stored into a 'metdata.csv' file, while the individual pprof files
+// are created per profile.
+func run(ctx context.Context) error {
+	client, err := cloudprofiler.NewExportClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+	log.Printf("Attempting to fetch %v profiles with a pageSize of %v for %v\n", *maxProfiles, *pageSize, *project)
+
+	// Initial request for the ListProfiles API
+	request := &pb.ListProfilesRequest{
+		Parent:    fmt.Sprintf("projects/%s", *project),
+		PageSize:  int32(*pageSize),
+		PageToken: *pageToken,
+	}
+
+	// create a folder for storing profiles & metadata
+	profilesDirName := fmt.Sprintf("profiles_%v", time.Now())
```

Comment:
nit i tried this out and it made pretty verbose pathnames that require escaping in the shell e.g. `profiles_2023-12-18 22:12:18.807095513 +0000 UTC m=+0.009132339`. Consider something a little more compact like `time.RFC3339` or unix timestamp

Reply:
I think, I will just use `time.Unix()`.